purchase_order_ubl/models/purchase.py

```diff
@@ -206,16 +220,23 @@ def get_ubl_lang(self):
     @api.multi
     def embed_ubl_xml_in_pdf(self, pdf_content):
         self.ensure_one()
-        doc_type = False
-        if self.state in self.get_rfq_states():
-            doc_type = 'rfq'
-        elif self.state in self.get_order_states():
-            doc_type = 'order'
+        doc_type = self.guess_ubl_doc_type()
         if doc_type:
             version = self.get_ubl_version()
             ubl_filename = self.get_ubl_filename(doc_type, version=version)
             xml_string = self.generate_ubl_xml_string(
                 doc_type, version=version)
+            self._ubl_check_xml_schema(
+                xml_string, self.document_name(doc_type), version=version)
             pdf_content = self.embed_xml_in_pdf(
                 xml_string, ubl_filename, pdf_content)
         return pdf_content
+
+    def guess_ubl_doc_type(self):
```

Comment:
Could you rename this method into `get_ubl_purchase_order_doc_type`? To make it consistent with https://github.com/OCA/edi/commit/4332084640b1933e6cf3742d4d697f2d2e8839a2#diff-54f706660be5cb2e069c3ec2c09f6dd9R214